tools/topology/topology2/avs-tplg/CMakeLists.txt

```diff
@@ -0,0 +1,59 @@
+# SPDX-License-Identifier: BSD-3-Clause
+
+# Array of "input-file-name;output-file-name;comma separated pre-processor variables"
+set(TPLGS
+# CAVS HDMI only topology with passthrough pipelines
+"cavs-passthrough-hdmi\;sof-hda-generic-idisp\;USE_CHAIN_DMA=true"
+# CAVS HDA topology with mixer-based pipelines for HDA and passthrough pipelines for HDMI
+"cavs-passthrough-hdmi\;sof-hda-generic\;HDA_CONFIG=mix,USE_CHAIN_DMA=true"
+# If the alsatplg plugins for NHLT are not available, the NHLT blobs will not be added to the
+# topologies below.
+"cavs-passthrough-hdmi\;sof-hda-generic-2ch\;\
+HDA_CONFIG=mix,NUM_DMICS=2,PREPROCESS_PLUGINS=nhlt,NHLT_BIN=nhlt-sof-hda-generic-2ch.bin,USE_CHAIN_DMA=true"
+"cavs-passthrough-hdmi\;sof-hda-generic-4ch\;\
+HDA_CONFIG=mix,NUM_DMICS=4,PDM1_MIC_A_ENABLE=1,PDM1_MIC_B_ENABLE=1,\
+PREPROCESS_PLUGINS=nhlt,NHLT_BIN=nhlt-sof-hda-generic-4ch.bin,USE_CHAIN_DMA=true"
+# CAVS SDW topology with passthrough pipelines
+"cavs-sdw\;cavs-sdw"
+
+# CAVS SDW with SRC gain and mixer support
+"cavs-sdw-src-gain-mixin\;cavs-sdw-src-gain-mixin"
+
+# SDW + HDMI topology with passthrough pipelines
+"cavs-sdw\;cavs-sdw-hdmi\;NUM_HDMIS=4"
+
+# CAVS SSP topology for TGL
+"cavs-nocodec\;sof-tgl-nocodec\;NUM_DMICS=4,PDM1_MIC_A_ENABLE=1,PDM1_MIC_B_ENABLE=1,\
+PREPROCESS_PLUGINS=nhlt,NHLT_BIN=nhlt-sof-tgl-nocodec.bin"
+)
+
+# generate ABI for IPC4
+add_custom_command(OUTPUT abi.conf
+	COMMAND > abi.conf  ${CMAKE_CURRENT_SOURCE_DIR}/../get_abi.sh ${SOF_ROOT_SOURCE_DIRECTORY} "ipc4"
+	DEPENDS ${SOF_ROOT_SOURCE_DIRECTORY}/src/include/kernel/abi.h
+)
+add_custom_target(cavs_abi_target
+	DEPENDS abi.conf
+)
```

Comment:
You shouldn't have to duplicate this in each subdirectory, you should be able to build a single abi.conf and use it everywhere. Are you not and why not?

Reply:
and fixed